src/spyglass/spikesorting/v1/recording.py

```diff
@@ -0,0 +1,730 @@
+        key["recording_id"] = generate_nwb_uuid(key["nwb_file_name"], "R", 6)
+        key["analysis_file_name"] = recording_nwb_file_name
+
+        # INSERT:
+        # - valid times into IntervalList
+        # - analysis NWB file holding processed recording into AnalysisNwbfile
+        # - entry into SpikeSortingRecording
+        IntervalList.insert1(
+            {
+                "nwb_file_name": key["nwb_file_name"],
+                "interval_list_name": key["recording_id"],
+                "valid_times": sort_interval_valid_times,
+            }
+        )
+        AnalysisNwbfile().add(key["nwb_file_name"], key["analysis_file_name"])
+        self.insert1(key)
+
+    @classmethod
+    def get_recording(cls, key: dict) -> si.BaseRecording:
+        """Get recording related to this curation as spikeinterface BaseRecording
+
+        Parameters
+        ----------
+        key : dict
+            primary key of SpikeSorting table
+        """
+
+        analysis_file_name = (cls & key).fetch1("analysis_file_name")
+        analysis_file_abs_path = AnalysisNwbfile.get_abs_path(
+            analysis_file_name
+        )
+        recording = se.read_nwb_recording(
+            analysis_file_abs_path, load_time_vector=True
+        )
+
+        return recording
+
+    @staticmethod
+    def _get_recording_timestamps(recording):
+        if recording.get_num_segments() > 1:
+            frames_per_segment = [0]
+            for i in range(recording.get_num_segments()):
+                frames_per_segment.append(
+                    recording.get_num_frames(segment_index=i)
+                )
+
+            cumsum_frames = np.cumsum(frames_per_segment)
+            total_frames = np.sum(frames_per_segment)
+
+            timestamps = np.zeros((total_frames,))
+            for i in range(recording.get_num_segments()):
+                timestamps[
+                    cumsum_frames[i] : cumsum_frames[i + 1]
+                ] = recording.get_times(segment_index=i)
+        else:
+            timestamps = recording.get_times()
+        return timestamps
+
+    def _get_sort_interval_valid_times(self, key: dict):
+        """Identifies the intersection between sort interval specified by the user
+        and the valid times (times for which neural data exist)
+
+        Parameters
+        ----------
+        key: dict
+            specifies a (partially filled) entry of SpikeSorting table
+
+        Returns
+        -------
+        sort_interval_valid_times: ndarray of tuples
+            (start, end) times for valid intervals in the sort interval
+
+        """
+        # FETCH:
+        # - sort interval
+        # - valid times
+        # - preprocessing parameters
+        sort_interval = (
+            IntervalList
+            & {
+                "nwb_file_name": key["nwb_file_name"],
+                "interval_list_name": key["sort_interval_name"],
+            }
+        ).fetch1("sort_interval")
```

Comment:
`sort_interval_name` is no longer a key in the `SpikeSortingRecordingSelection`. I think this key should be `interval_list_name` now.  This applies in the next fetch as well

`sort_interval` isn't a key in IntervalList

Reply:
Thanks, I will fix this.

Reply:
Fixed